packages/cosmic-swingset/lib/launch-chain.js

```diff
@@ -182,9 +182,19 @@ export async function launch(
     );
   }
 
-  const [savedHeight, savedActions, savedChainSends] = JSON.parse(
-    storage.get(SWING_STORE_META_KEY) || '[0, [], []]',
+  const [initSavedHeight, savedActions, savedChainSends] = JSON.parse(
+    storage.get(SWING_STORE_META_KEY) || '[-1, [], []]',
   );
+
+  let savedHeight = initSavedHeight;
+  if (savedHeight < 0) {
+    // We need to fully bootstrap the chain before we can be open to receive
+    // outside messages.
+    await controller.run();
+    savedHeight = 0;
+    await saveOutsideState(savedHeight, savedActions, savedChainSends);
```

Comment:
This has an `await` in a nested `if`, which is not OK.  Move the `if` into to a separate async function, where the negative of it returns and then the rest continues.

I notice that file has other `await` rule violations. ``await` is only allowed at the top level of a function or module, not inside any control constructs. Otherwise you end up with code that might be invoked in the same turn or different turns, depending. That's an async bug farm.

Reply:
I cleaned up this one case.  Will file an issue for other cleanups.